src/main/java/seedu/address/logic/commands/EditMeetingCommand.java

```diff
@@ -0,0 +1,280 @@
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_MEETING = "This meeting already exists in the address book.";
+    public static final String MESSAGE_INVALID_TIME = "Meeting end time should be later meeting start time";
+
+    private final Index index;
+    private final EditMeetingCommand.EditMeetingDescriptor editMeetingDescriptor;
+
+    /**
+     * @param index                of the person in the filtered meeting list to edit
+     * @param editMeetingDescriptor details to edit the meeting with
+     */
+    public EditMeetingCommand(Index index, EditMeetingDescriptor editMeetingDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editMeetingDescriptor);
+
+        this.index = index;
+        this.editMeetingDescriptor = new EditMeetingCommand.EditMeetingDescriptor(editMeetingDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Meeting> lastShownList = model.getFilteredMeetingList();
+        List<Contact> contactList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MEETING_DISPLAYED_INDEX);
+        }
+
+        Meeting meetingToEdit = lastShownList.get(index.getZeroBased());
+        Meeting editedMeeting = createEditedMeeting(meetingToEdit, editMeetingDescriptor, contactList);
+
+        if (!meetingToEdit.isSameMeeting(editedMeeting) && model.hasMeeting(editedMeeting)) {
+            throw new CommandException(MESSAGE_DUPLICATE_MEETING);
+        }
+
+        model.setMeeting(meetingToEdit, editedMeeting);
+        model.updateFilteredMeetingList(PREDICATE_SHOW_ALL_MEETINGS);
+        model.commitAddressBook();
+        return new CommandResult(String.format(MESSAGE_EDIT_MEETING_SUCCESS, editedMeeting));
+    }
+
+    /**
+     * Creates and returns a {@code Meeting} with the details of {@code meetingToEdit}
+     * edited with {@code editMeetingDescriptor}.
+     */
+    private static Meeting createEditedMeeting(Meeting meetingToEdit, EditMeetingDescriptor editMeetingDescriptor,
+                                               List<Contact> contactList) throws CommandException {
+        assert meetingToEdit != null;
+
+        MeetingName updatedMeetingName = editMeetingDescriptor.getMeetingName().orElse(meetingToEdit.getName());
+        MeetingDate updatedMeetingDate = editMeetingDescriptor.getMeetingDate().orElse(meetingToEdit.getDate());
+        StartTime updatedStartTime = editMeetingDescriptor.getStartTime().orElse(meetingToEdit.getStartTime());
+        EndTime updatedEndTime = editMeetingDescriptor.getEndTime().orElse(meetingToEdit.getEndTime());
+
+        if (updatedStartTime.isAfter(updatedEndTime)) {
+            throw new CommandException(MESSAGE_INVALID_TIME);
+        }
```

Comment:
I think for this part we should use `!updatedStartTime.isBefore(updatedEndTime)` or `updatedStartTime.isAfter(updatedEndTime) && updatedStartTime == updatedEndTime` to throw exception for the case  where startTime is equal to endTime!

Reply:
```
public boolean isAfter(Time time) {
        return !(this.value.compareTo(time.value) < 0);
 }
```
Actually `Time::isAfter` takes into account the `updatedStartTime == updatedEndTime` case